jobs/elasticsearch/templates/bin/post-start

```diff
@@ -1,6 +1,26 @@
 #!/bin/bash
 # Inspired by https://github.com/cloudfoundry/exemplar-release/blob/master/jobs/paragon/templates/post-start
 # If a command fails, exit immediately
+set +e
+
+echo "== AssociateIamInstanceProfile =="
```

Comment:
Why are both access key and instance profile required?

Reply:
sorry late reply.
I checked elasticsearch s3 repo plugin document.
https://www.elastic.co/guide/en/elasticsearch/plugins/current/repository-s3-usage.html
>  The repository defaults to using ECS IAM Role or EC2 IAM Role credentials for authentication. 

Actually executing with out "AssociateIamInstanceProfile" will give an error stating that the permission is insufficient.

Reply:
My question is why are you using instance profile rather than putting  ECS IAM Role or EC2 IAM Role to the account (`AWS_ACCESS_KEY_ID `/`AWS_ACCESS_SECRET_KEY `) ?

Reply:
By the way, I was able to create a snapshot with 

``` yaml
- type: replace
  path: /instance_groups/name=elasticsearch-master/jobs/name=elasticsearch/properties/elasticsearch/plugins?/-
  value: 
    epository-s3: /var/vcap/packages/repository-s3/repository-s3.zip
- type: replace
  path: /instance_groups/name=elasticsearch-master/jobs/name=elasticsearch/properties/elasticsearch/plugin_install_opts?
  value: 
  - --batch
- type: replace
  path: /instance_groups/name=elasticsearch-master/jobs/-
  value:
    name: repository-s3
    release: elasticsearch
- type: replace
  path: /instance_groups/name=elasticsearch-master/jobs/name=elasticsearch/properties/elasticsearch/jvm_options?
  value: 
  - -Des.allow_insecure_settings=true
```


```
curl -s -XPUT aaa.bbb.ccc.ddd:9200/_snapshot/foo -H 'Content-Type: application/json' -d '{
  "type": "s3",
  "settings": {
    "bucket": "essnapshot3",
    "region": "ap-northeast-1",
    "endpoint": "https://s3-ap-northeast-1.amazonaws.com",
    "access_key": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "secret_key": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
  }
}' | jq .
```